pages/api/transactions/index.js

```diff
@@ -1,39 +1,62 @@
 import dbConnect from "@/db/connect";
 import Transaction from "@/db/models/Transaction";
+import { getServerSession } from "next-auth/next";
+import { authOptions } from "../auth/[...nextauth]";
+import { getToken } from "next-auth/jwt";
 
 export default async function handler(request, response) {
+  const session = await getServerSession(request, response, authOptions);
+  const token = await getToken({ req: request });
+  const userId = token?.sub;
+
+  console.log(token);
   await dbConnect();
+
   if (request.method === "GET") {
-    try {
-      const transactions = await Transaction.find();
+    /* const transactions = await Transaction.find();
+    response.status(200).json(transactions);
+    return; */
+
+    if (session) {
+      const transactions = await Transaction.find({ owner: token.email });
       response.status(200).json(transactions);
       return;
-    } catch (error) {
-      response.status(500).json({ message: "Error finding transactions" });
+    } else {
+      const transactions = await Transaction.find({ owner: "default" });
+      response.status(200).json(transactions);
       return;
     }
   }
 
- if (request.method === "POST") {
+  if (request.method === "POST") {
     try {
-      const { name, amount, category, type, date } = request.body || {};
+      if (session) {
+        const { name, amount, category, type, date } = request.body || {};
 
-     // Validation 
-      if (!name || !amount || !category || !type || !date) {
-        return response.status(400).json({ message: "Bitte alle Felder ausfüllen." });
-      }
+        // Validation
+        if (!name || !amount || !category || !type || !date) {
+          return response
+            .status(400)
+            .json({ message: "Bitte alle Felder ausfüllen." });
+        } else {
+          response.status(401).json({ status: "Not authorized" });
```

Comment:
'Not authorized' is wrong here. Please check and modify the code structure. Look: You run into this else if the data is valid. The only reason, why it works is that there is no 'return' keyword in this line. So the following code will also be executed.